### ISSUE TITLE
uri! can't be used on routes with params whose names begin with _

### ISSUE THREAD

**Issue by jebrosen** [SergioBenitez/Rocket#603]
Rocket version: `master`
Operating system: Arch Linux; rustc 1.27.0-nightly (0b72d48f8 2018-04-10)

When a uri!() macro is used on a route with parameters named with an initial _, there is an error in the macro expansion:

```
error: invalid format string: expected `'}'`, found `'_'`
 --> examples/uri_fail/src/main.rs:1:1
  |
1 | #![feature(plugin, decl_macro)]
  | ^
...
9 |     uri!(fail: "test");
  |     ------------------- in this macro invocation
```

Code to reproduce:
```
#![feature(plugin, decl_macro)]
#![plugin(rocket_codegen)]

extern crate rocket;

#[get("/")]
fn hello() -> &'static str {
    uri!(works: "test");
    uri!(fail: "test");
    "whatever"
}

#[get("/works/<used>")]
fn works(used: String) -> String {
    // can uri! to this one
    used
}

#[get("/fail/<_unused>")]
fn fail(_unused: String) -> &'static str {
    //can't uri! to this one
    "text"
}

fn main() {
    rocket::ignite().mount("/", routes![hello, works, fail]).launch();
}
```

`ROCKET_CODEGEN_DEBUG=1` reveals the cause:
```
--> codegen/src/macros/uri.rs:172
Emitting URI expression: expr(4294967295: {
    let _unused =
        <String as
            ::rocket::http::uri::FromUriParam<_>>::from_uri_param("test");
    ::rocket::http::uri::Uri::from(format!("/fail/{_unused}" , _unused = &
                                           _unused as &
                                           ::rocket::http::uri::UriDisplay ,))
})
```

Apparently the format! macro can't be used with leading underscores in names, which is what the compiler suggests to do with unused variables.

I believe Rocket should either work around this or print a better diagnostic.

**Comment by jebrosen** [comment]
A possible workaround, assuming it works as intended: prepending some set of known good characters to the format! params.

Doctored up examples:

```
--> codegen/src/macros/uri.rs:172
Emitting URI expression: expr(4294967295: {
    let _unused =
        <String as
            ::rocket::http::uri::FromUriParam<_>>::from_uri_param("test");
    ::rocket::http::uri::Uri::from(format!("/fail/{p__unused}" , p__unused = &
                                           _unused as &
                                           ::rocket::http::uri::UriDisplay ,))
})
```

```
--> codegen/src/macros/uri.rs:172
Emitting URI expression: expr(4294967295: {
    let used =
        <String as
            ::rocket::http::uri::FromUriParam<_>>::from_uri_param("test");
    ::rocket::http::uri::Uri::from(format!("/fail/{p_used}" , p_used = &
                                           used as &
                                           ::rocket::http::uri::UriDisplay ,))
})
```

**Comment by SergioBenitez** [comment]
Fixed in 648eb1a5ebd50099e90ef26f1f641f2b685175ec.
